code/modules/mob/language/language.dm

```diff
@@ -158,7 +158,12 @@
 
 // Can we speak this language, as opposed to just understanding it?
 /mob/proc/can_speak(datum/language/speaking)
-	return (speaking.can_speak_special(src) && (universal_speak || (speaking && speaking.flags & INNATE) || speaking in src.languages))
+//Prevents someone from speaking a null language.
+	if(speaking)
+		return (speaking.can_speak_special(src) && (universal_speak || (speaking && speaking.flags & INNATE) || speaking in src.languages))
```

Comment:
I know it's not part of this PR, but you're touching the line anyway.

```(speaking && speaking.flags & INNATE)``` should really be ````(speaking && (speaking.flags & INNATE))```` to avoid && being evaluated before &.